src/sql/workbench/services/connection/browser/connectionDialogService.ts

```diff
@@ -275,10 +276,13 @@ export class ConnectionDialogService implements IConnectionDialogService {
 			} else if (connectionResult && connectionResult.errorHandled) {
 				this._connectionDialog.resetConnection();
 				this._logService.debug(`ConnectionDialogService: Error handled and connection reset - Error: ${connectionResult.errorMessage}`);
-			} else {
+			} else if (connectionResult.errorCode !== Constants.sqlPasswordErrorCode) {
```

Comment:
it seems like we should test `connectionResult.errorCode === Constants.sqlPasswordErrorCode` and do the password change.  And leave the default case the same as before.  i.e. making password change the default doesn't make sense from readability to me.